### ISSUE TITLE
CJK Unified Ideographs (Unicode block) range in examples/booleansearchparser.py

### ISSUE THREAD

**Issue by tc-yu** [pyparsing/pyparsing#339]
https://github.com/pyparsing/pyparsing/blob/b5b1fdd1c0a3dfffbbad206e885d59e6015015a8/examples/booleansearchparser.py#L105

According to  https://en.wikipedia.org/wiki/CJK_Unified_Ideographs_(Unicode_block) should probably be `[int("4E00", 16), int("9FFF", 16)]`?

**Comment by ptmcg** [comment]
Looks right. Care to submit a PR? Please be sure to add a test using characters in this range. The test should fail with the old code but pass with the updated code. (Searching for this character `亀` will do it.)

**Comment by tc-yu** [comment]
I'll try to submit one. Just got into PyParsing a few days ago, still figuring how stuff works.

**Comment by ptmcg** [comment]
Glad to hear you are digging in to learning pyparsing, but you shouldn't have to learn too much pyparsing for this one. Just update the line you cited, and add a test string and expected matches to the set of tests at the bottom of that same script. Without your change, you'll probably get a parsing error, which is the issue your change is fixing. Then add your change and the test will run.

**Comment by tc-yu** [comment]
https://github.com/pyparsing/pyparsing/blob/b5b1fdd1c0a3dfffbbad206e885d59e6015015a8/examples/booleansearchparser.py#L155-L156

Upon further investigation, the logic of adding characters to alphabets also need an update, the current version are not adding the last character of each Unicode block, e.g. `ヿ` (U+30FF) to the list of alphabets

**Comment by ptmcg** [comment]
This non-Western support was what really inspired my addition of the `pyparsing_unicode` namespace class. I wonder if these alphabet ranges could be replaced with usages of the character ranges in that class now. Would look like this:

```
from pyparsing import (
    Word,
    Keyword,
    Group,
    Forward,
    Suppress,
    OneOrMore,
    oneOf,
    pyparsing_unicode as ppu
)

...

        alphabet = ppu.Latin1.alphanums

        # support for non-western alphabets
        class NonWesternSupport(ppu.Thai, ppu.Cyrillic, ppu.CJK, ppu.Arabic):
            pass
        alphabet += NonWesternSupport.printables
```

(Those ranges correctly include the last character in each range, btw.)

**Comment by tc-yu** [comment]
Need to experiment with `ppu`, kept the original add character approach in PR.

**Comment by ptmcg** [comment]
Released in 3.0.7
